src/hooks/useFlags/index.ts

```diff
@@ -0,0 +1,33 @@
+import { useAllEnvFlags, useEnvFlags } from "./env";
+import { useAllFlagsmithFlags, useFlagsmithFlags } from "./flagsmith";
+import { FlagsResults, FlagWithName } from "./types";
+
+export const useFlags = <T extends readonly string[]>(
+  flags: readonly [...T],
+  traits?: string[],
+): FlagsResults<T> => {
+  const flagsmithFlags = useFlagsmithFlags(flags, traits);
```

Comment:
you shouldn't even call this when service is not enabled

Reply:
Picking the service is not a runtime decision. You make that decision only once, when you build the project. Knowing this you can pick which hook you need to use, 

For example:

```
const useSpecific = HAS_SERVICE ? useFlagsmithFlags : useEnvFlags;

const useFlags = () => {
 const flags = useSpecific()
 // ...
}
```

The `HAS_SERVICE` once is built, it's always true, so you are not switching hook calling within the runtime.

An even better option is to not use their hooks, but fetch flags from their API directly, then you have full control of fetching something or not (because you cannot switch hooks conditionally within the runtime)

Reply:
Furthermore, imagine now we change to service, so you need to amend all of your hooks to use a different hook, using even option 1, you centralize it